### PR TITLE
Add forms examples

### PR DESCRIPTION
**Pull request by lilyvc**
Adds some examples of forms pattern to storybook.

### CODE REVIEW

File: packages/core/stories/patterns/forms/forms.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,324 @@
...
+            <Text style={{ whiteSpace: "nowrap" }}>US Dollar</Text>
+          </StackLayout>
+        </FormField>
+      </GridItem>
+      <GridItem colSpan={4}>
+        <hr></hr>
+      </GridItem>
+      <GridItem colSpan={1}>
+        <FormField>
+          <FormFieldLabel>Client directed request?</FormFieldLabel>
```
Comment (origami-z) [inline packages/core/stories/patterns/forms/forms.stories.tsx:130]: Span more to match design?

File: packages/core/stories/patterns/forms/forms.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,324 @@
...
+        <FormField>
+          <FormFieldLabel>Expected total annual valurs</FormFieldLabel>
+          <StackLayout direction="row" align="center">
+            <Input placeholder="e.g., 100000" />
+            <Text style={{ whiteSpace: "nowrap" }}>US Dollar</Text>
+          </StackLayout>
+        </FormField>
+      </GridItem>
+      <GridItem colSpan={4}>
+        <hr></hr>
```
Comment (origami-z) [inline packages/core/stories/patterns/forms/forms.stories.tsx:126]: Different color to design? Worth thinking implement a component with primary/secondary/tertiary separable token.

File: packages/core/stories/patterns/forms/forms.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,324 @@
...
+            validationStatus={isError ? "error" : undefined}
+          />
+          <FormFieldHelperText>Helper text</FormFieldHelperText>
+        </FormField>
+      </GridItem>
+    </GridLayout>
+  );
+};
+
+export const Compact = () => {
```
Comment (origami-z) [inline packages/core/stories/patterns/forms/forms.stories.tsx:279]: Pattern doc shows right aligned by default? Readonly is left aligned? 
Could match more closely to the pattern image

**Review by origami-z — COMMENTED** [review]
Add a link from pattern page to storybook?

File: packages/core/stories/patterns/forms/forms.stories.tsx (hunk 2 above)
Comment (origami-z) [reply packages/core/stories/patterns/forms/forms.stories.tsx:126]: Will have proper design done in #3320

Comment (lilyvc) [reply packages/core/stories/patterns/forms/forms.stories.tsx]: changed to use a border, can be updated after design

**Review by origami-z — DISMISSED** [review]

File: site/docs/patterns/forms.mdx [hunk 4]
```diff
@@ -5,11 +5,13 @@ aliases:
   - /salt/patterns/forms
 data:
   components: ["Button", "Checkbox", "Combo box", "Dropdown", "Form field", "Radio button"]
-  relatedPatterns: ["Button bar"]
+  relatedPatterns: ["Button bar"] 
+  http://localhost:6006/?path=/story/patterns-forms--standard-layout
```
Comment (joshwooding) [inline site/docs/patterns/forms.mdx:9]:
```suggestion
```

Comment (lilyvc) [reply site/docs/patterns/forms.mdx]: oh yeah good spot, there's an extra space as well which I've removed

**Review by origami-z — APPROVED** [review]

**Review by lukeac123 — APPROVED** [review]

**Pull request merged**